improved testing/formatting section

alexanderLinear opened the pull request:
This pull request is purely about documentation improvements.
It first re-orders and slightly improves formatting so that the overall meaning and usability of the referenced tooling is eased.

alexanderLinear committed:
improved testing/formatting section

tfoote commented on CONTRIBUTING.md:
```diff
@@ -295,14 +295,19 @@ How to submit pull requests
 
-To run the tests run ``nosetests`` in the root of the repository.
-These tests require several dependencies that can be installed either from the ROS repositories or via pip(list built based on the content of [.travis.yaml](https://github.com/ros/rosdistro/blob/master/.travis.yml):
+The following modules/packages are needed and will see install by avobe comment.
```
```suggestion
The following modules/packages are needed and will see install by above comment.
```

alexanderLinear replied:
thanks, used it with a bit more of needed change.

tfoote reviewed (changes requested):
Thanks for contributing the extra details. One small fixup requested though.

alexanderLinear committed:
updated according to reviews

alexanderLinear commented:
> Also please replace the body of the issue/PR with a real description to enable easier review for others in the future.

done.

tfoote reviewed (approved):
Thanks for the attention. @quarkytale Can you try this out and if it works for you merge?

quarkytale commented on CONTRIBUTING.md:
````diff
@@ -295,14 +295,19 @@ How to submit pull requests
 python3 -m pip install -r test/requirements.txt
 ```
 
-To run the tests run ``nosetests`` in the root of the repository.
````
This line made sense to me after installing dependencies, and after the list of installed modules.

quarkytale commented on CONTRIBUTING.md:
```diff
@@ -295,14 +295,19 @@ How to submit pull requests
-If you want to run the tests before submitting, first install the dependencies. Using `pip` is recommended.
+#### Testing
+
+To run the tests run ``nosetests`` (nose tests) in the root of the repository.
```
A separate section is a good idea but maybe some other intro line or about nose https://nose.readthedocs.io/en/latest/index.html? It would be better for uses to have the command after installing dependencies.

quarkytale commented on CONTRIBUTING.md:
```diff
@@ -295,14 +295,19 @@ How to submit pull requests
+
+To run the tests run ``nosetests`` (nose tests) in the root of the repository.
+
+Note: There's a [known issue](https://github.com/disqus/nose-unittest/issues/2) discovered [here](https://github.com/ros/rosdistro/issues/16336) that most tests won't run if you have the python package `nose-unitttest` installed.
```
```suggestion
Note: There's a [known issue](https://github.com/disqus/nose-unittest/issues/2) discovered [here](https://github.com/ros/rosdistro/issues/16336) that most tests won't run if you have the python package `nose-unittest` installed.
```

quarkytale commented on CONTRIBUTING.md:
```diff
@@ -295,14 +295,19 @@ How to submit pull requests
 
-To run the tests run ``nosetests`` in the root of the repository.
-These tests require several dependencies that can be installed either from the ROS repositories or via pip(list built based on the content of [.travis.yaml](https://github.com/ros/rosdistro/blob/master/.travis.yml):
+The following modules/packages are needed and will see install by the above command.
```
```suggestion
The following modules/packages are needed and will be installed by the above command.
```

quarkytale commented on CONTRIBUTING.md:
```diff
@@ -314,6 +319,6 @@ These tests require several dependencies that can be installed either from the R
+#### Formatting
 
-Note: There's a [known issue](https://github.com/disqus/nose-unittest/issues/2) discovered [here](https://github.com/ros/rosdistro/issues/16336) that most tests won't run if you have the python package `nose-unitttest` installed.
+There is further the tool ``rosdistro_reformat`` which will fix most formatting errors such as alphabetization and correct formatting. This tool takes an ``index`` parameter that is a required command line argument. With option ``-n`` a dry run is possible. 
```
```suggestion
There is a tool ``rosdistro_reformat`` which will fix most formatting errors such as alphabetization and correct formatting. This tool takes an ``index`` parameter that is a required command line argument. With option ``-n`` a dry run is possible. 
```

AlexanderStohr committed:
eliminate deprecation warning on escapes by using raw string

AlexanderStohr committed:
eliminate deprecation warning with escape

https://stackoverflow.com/questions/50504500/deprecationwarning-invalid-escape-sequence-what-to-use-instead-of-d

alexanderLinear committed:
Merge pull request #1 from AlexanderStohr/patch-4

eliminate deprecation warning on escapes by using raw string

alexanderLinear committed:
Merge pull request #2 from AlexanderStohr/patch-5

eliminate deprecation warning with escape

quarkytale commented:
Hi @AlexanderStohr would you like to address some of my previous comments before I give it another review

nuclearsandwich commented:
@alexanderLinear are you still interested in addressing this feedback and updating this PR?
If so, great! If not, could you please close it.

alexanderLinear commented:
> @alexanderLinear are you still interested in addressing this feedback and updating this PR? If so, great! If not, could you please close it.

sorry, was busy in recent times.
approaching a long weekend with only few other dates. will try to go on with it again then.

emersonknapp commented:
Closing as stale per guidelines. If you still wish to contribute this change, feel free to open a new PR
